### DISCUSSION TITLE
Got it working, here are tips (Arch Linux)

### DISCUSSION THREAD

**Discussion by feldi12** [cryinkfly/Autodesk-Fusion-360-for-Linux#415] [Show and tell]
Hi all,
I got the Fusion360 working under my arch linux, and below are problems and solutions I came across. Remember to run all the wine commands in proper prefix where you set install dir when running crinkflys install script.

- At first try installation silently failed half way. There were parts of Fusion installed, but no Fusion360.exe to be found. Running the script 2nd time seems to have helped.
- The `box-run.sh` fails on my system, but running `WINEPREFIX='<install dir' FUSION_IDSDK=false WINEDEBUG=-all,-d3d wine '<Fusion360.exe path>' `works. Fusion360.exe should be in yours C:\user\<username>\AppData folder in wine prefix. Run find` -iname fusion360.exe` to find it
- I didn't get browser running in wine, the activation opened in my linux browser instead. This means, that URI from browser to Autodesk Identity Manager fails, one need to help it manually. To do so, after you authenticate into your autodesk account in your native linux browser,  copy link of the 'go to product' button. This should start with  `adskidmgr://` and a lot of text after that. Then find AdskIdentityManager.exe in your fusion360 wine prefix, navigate there and run `wine  AdskIdentityManager.exe '<insert the link here>'`. The ' ' are important and remember about the prefix.
- In my case the startup window of Fusion360 was frozen after running the above, but once I killed it with xkill and started again, Fusion360 was already logged into my account.

Lastly, I had some bad performance issues. I'm running nvidia GT1030 with nvidia proprietary driver (dkms package) and Fusion would not smoothly render anything. Even sketching a rectangle was a slideshow. Also there was high Xorg usage at times. This was solved by adding `options nvidia_drm modeset=1 fbdev=1` to `/etc/modprobe.d/nvidia.conf` as advised in [Arch wiki ](https://wiki.archlinux.org/title/NVIDIA#DRM_kernel_mode_setting) (needs system restart). Now everything runs smoothly (loads a bit slow - about 30 seconds).

Problems still unsolved:

- There seems to be some dialogs that you can't run (Fusion freezes). One I found, was trying to get into preferences in your account icon.
- A lot of wine processes are left running after you close Fusion360. I kill those with ``killall -r ".*\.exe" (this will kill any process ending with .exe, you might need to trim this to your needs).

Hope this helps someone
Cheers

Setup: 
Intel(R) Xeon(R) CPU E3-1220 v5 @ 3.00GHz
16GB of RAM
nvidia GT1030, nvidia proprietary driver
Arch Linux LTS kernel branch

**Comment by jeromio** [comment]
This looked so promising, but sadly, no joy for me. FWIW, I had Fusion working a few months back just using @cryinkfly scripts. It's what finally pushed me off W11 (onto Mint). Then I guess there was a wine update and I didn't pay close enough attention. A few weeks ago I started to just get a black screen. So I've been in a spiral trying to get Fusion working again. I've downgraded to 10.13 which fixes the black screen, but gets me stuck at the sign in screen. This post got me deeper into the weeds 😁.

I was able to execute the AdskIdentityManager.exe, but it failed with a exception. Fixed that with regedit from issue #432 . 

The link I have in my "go to product" is actually sideways in the sign in web page, it's a hidden  form element. It's in double quotes and also only has a single slash, eg 
`<input type="hidden" id="product-sign-in-link" value="adskidmgr:/login?code=ue1c2.1bD..............so many chars........=">`
SO, after Fusion pops the sign in screen, I tried executing wine AdskIdentityManager.exe then with the adskidmgr link in single quotes, in double quotes, with an additional (more correct looking) double slash, etc. No change, still just stuck on "Sign in" screen.

I also tried making a mini bash script:
```
#!/usr/bin/env bash
wine /home/jeromio/.autodesk_fusion/wineprefixes/default/drive_c/Program\ Files/Autodesk/webdeploy/production/37a4fdb912fcb7645f48036186017c0cc6730354/Autodesk\ Identity\ Manager/AdskIdentityManager.exe %1
```
and then I choose that when the browser tries to open the link. Same sad result.
